File: modules/swagger-core/src/test/scala/BeanValidatorTest.scala

import io.swagger.converter.ModelConverters
import io.swagger.models.properties.{DoubleProperty, IntegerProperty, StringProperty}
import models._
import org.junit.runner.RunWith
import org.scalatest.{FlatSpec, Matchers}
import org.scalatest.junit.JUnitRunner


@RunWith(classOf[JUnitRunner])
class BeanValidatorTest extends FlatSpec with Matchers {
  it should "read bean validations" in {
    val schemas = ModelConverters.getInstance().readAll(classOf[BeanValidationsModel])
    val model = schemas.get("BeanValidationsModel")
    val properties = model.getProperties()

    val age = properties.get("age").asInstanceOf[IntegerProperty]
    age.getMinimum() should be(13.0)
    age.getMaximum() should be(99.0)

    val password = properties.get("password").asInstanceOf[StringProperty]
    password.getMinLength() should be(6)
    password.getMaxLength() should be(20)

    val minBalance = properties.get("minBalance").asInstanceOf[DoubleProperty]
    minBalance.getExclusiveMinimum().booleanValue() should be(true)

    val maxBalance = properties.get("maxBalance").asInstanceOf[DoubleProperty]
    maxBalance.getExclusiveMaximum().booleanValue() should be(true)
  }
}

File: modules/swagger-core/src/test/java/io/swagger/BeanValidatorTest.java
package io.swagger;

import io.swagger.converter.ModelConverters;
import io.swagger.models.BeanValidationsModel;
import io.swagger.models.Model;
import io.swagger.models.properties.DoubleProperty;
import io.swagger.models.properties.IntegerProperty;
import io.swagger.models.properties.Property;
import io.swagger.models.properties.StringProperty;
import org.testng.Assert;
import org.testng.annotations.Test;

import java.util.Map;

public class BeanValidatorTest {

    @Test(description = "read bean validations")
    public void readBeanValidatorTest() {
        final Map<String, Model> schemas = ModelConverters.getInstance().readAll(BeanValidationsModel.class);
        final Model model = schemas.get("BeanValidationsModel");
        final Map<String, Property> properties = model.getProperties();

        final IntegerProperty age = (IntegerProperty) properties.get("age");
        Assert.assertEquals(age.getMinimum(), 13.0);
        Assert.assertEquals(age.getMaximum(), 99.0);

        final StringProperty password = (StringProperty) properties.get("password");
        Assert.assertEquals((int) password.getMinLength(), 6);
        Assert.assertEquals((int) password.getMaxLength(), 20);

        final StringProperty email= (StringProperty) properties.get("email");
        Assert.assertEquals((String) email.getPattern(), "(.+?)@(.+?)");

        final DoubleProperty minBalance = (DoubleProperty) properties.get("minBalance");
        Assert.assertTrue(minBalance.getExclusiveMinimum());

        final DoubleProperty maxBalance = (DoubleProperty) properties.get("maxBalance");
        Assert.assertTrue(maxBalance.getExclusiveMaximum());
    }
}
